src/transformers/utils/model_parallel_utils.py

```diff
@@ -15,6 +15,154 @@
 
 from math import ceil
 
+from .. import BertPreTrainedModel, GPT2PreTrainedModel, TransfoXLPreTrainedModel
+
+
+# ie. nn.Linear(3 * dim, dim)
+# only defined the models that have fused attention layer.
+FUSED_ATTENTION_MAPPING = {
+    GPT2PreTrainedModel: {"attn.c_attn": 3, "crossattention.c_attn": 2},
+    TransfoXLPreTrainedModel: {"qkv_net": 3},
+}
+
+# ie. nn.Linear(out_dim, in_dim) or Conv1D()
+# only defined the models that have reversed parameters.
+REVERSED_PARAM_MAPPING = {
+    GPT2PreTrainedModel: ["attn", "crossattention", "mlp"],
+    TransfoXLPreTrainedModel: ["qkv_net"],
+}
+
+# All the mapping for tensor parallelism
+TENSOR_PARALLEL_MAPPING = {
```

Comment:
I wrote the mappings of the two most representative models. Let's expand gradually.